### ISSUE TITLE
distribution tag not applied

### ISSUE THREAD

**Issue by lbrigman124** [nebula-plugins/gradle-ospackage-plugin#144]
In building rpm packages on Centos 6 with gradle 2.9 
I specify 
ospackage {
  distribution = 'el6'
}
or in 'task buildRpm'
but it doesn't add the tag.
I also tried using what rpm macros calls it 'dist' but that fails as it won't compile. 
Same result with buildRpm.distribution and buildRpm.dist

**Comment by sghill** [comment]
I tried to reproduce this with v4.3.0 on gradle 3.2.1, but it appears to be working. #223 adds to the test suite to ensure it continues to work going forward.

```gradle
// issue-144-pkg/build.gradle
plugins {
    id 'java'
    id 'nebula.ospackage' version '4.3.0'
}

ospackage {
    distribution = 'el6'
    release '1'
    version '1.0.0'
    from (jar.outputs.files) {
        into '/lib'
    }
}
```

```shell
$ rpm -qp build/distributions/issue-144-pkg-1.0.0-1.noarch.rpm --qf '%-15{NAME}%{DISTRIBUTION}'
issue-144-pkg  el6
```

**Comment by lbrigman124** [comment]
It doesn't put the distribution tag into the name which is typical for rpm
packages and what I would expect.

**Comment by sghill** [comment]
Thanks for the followup @lbrigman124. 

The name is generated by the upstream library [redline][rl]. Indeed it isn't taking distribution into account. If you'd like to pursue a fix there, [`org.redline_rpm.Builder#build(java.io.File directory)`][block] should be a good starting point.

[rl]: https://github.com/craigwblake/redline
[block]: https://github.com/craigwblake/redline/blob/75ec29838531389afab9e3fe78b3befbc57dff7f/src/main/java/org/redline_rpm/Builder.java#L1233-L1241

**Comment by sghill** [comment]
I dove into this a bit more today and thought it'd be beneficial to circle back.

Redline and this plugin both use the same naming convention of `name-version-release.architecture.rpm`, which matches the [docs at rpm.org][rpm]. ospackage is [assembling the archive name][assemble] and passing it to another method in redline. 

We don't have a need for a configurable archive name, but would be open to a pull request. Writing a task in your build script that renames the output the way you'd like it is another option.

[assemble]: https://github.com/nebula-plugins/gradle-ospackage-plugin/blob/07b3b25abd9901bd4a2b78e6b8c931de5a88a3ab/src/main/groovy/com/netflix/gradle/plugins/rpm/Rpm.groovy#L39-L46

[rpm]: http://rpm.org/max-rpm-snapshot/ch-rpm-file-format.html

**Comment by lbrigman124** [comment]
I did some checking of the packages on RedHat CentOS and Scientific Linux.
The release information has the distro tag attached.
That is done automatically by the rpm scripts if the .dist macro is set.

I was able to test this yesterday by appending the distribution variable to
the release variable.  This causes the system to properly name the package
and produce a
package that has the release info matching the expectation from packages
built with the rpmbuild suite.

Maybe the only thing that is needed here is to add some notes/docs about
doing distribution specific packages via the method noted above.
